src/hb-ot-glyf-table.hh

```diff
@@ -211,20 +211,14 @@ struct glyf
   _fix_component_gids (const hb_subset_plan_t *plan,
 		       hb_bytes_t glyph)
   {
-    OT::glyf::CompositeGlyphHeader::Iterator iterator;
-    if (OT::glyf::CompositeGlyphHeader::get_iterator (&glyph,
-						      glyph.length,
-						      &iterator))
-    {
-      do
-      {
-	hb_codepoint_t new_gid;
-	if (!plan->new_gid_for_old_gid (iterator.current->glyphIndex,
-					&new_gid))
-	  continue;
-	((OT::glyf::CompositeGlyphHeader *) iterator.current)->glyphIndex = new_gid;
-      } while (iterator.move_to_next ());
-    }
+    + OT::glyf::get_composite_iterator (glyph)
```

Comment:
> Our naming scheme is typically just iter() returning an iterator. In this case, I'd say composite_iter... But also, don't we have a Glyph object?

Just it shouldn't be inside CompositeGlyph itself as clang can not accept using a non-completed struct inside itself.

> Do evolve slowly instead of making the PR bigger and bigger.

Reaching to build-able state on every commit makes this hard (but I intended to add hb_foreach), for example on above clang issue.

Reply:
> Do evolve slowly instead of making the PR bigger and bigger.

Ok. Now I see some parts of it is separable, will upload them separately if you want to.

Reply:
> > Our naming scheme is typically just iter() returning an iterator. In this case, I'd say composite_iter... But also, don't we have a Glyph object?
> 
> Just it shouldn't be inside CompositeGlyph itself as clang can not accept using a non-completed struct inside itself.

Can you elaborate?  I would think we should do `glyph_bytes.as<CompositeGlyph>.iter ()`...

Reply:
Should be sanitized or something... Or maybe iter needs the full bytes indeed.  Anyway.  Do what you can and we can evolve later.